superset/assets/javascripts/components/ColumnTypeLabel.jsx

```diff
@@ -2,16 +2,20 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 const propTypes = {
-  type: PropTypes.string.isRequired,
+  type: PropTypes.string,
 };
 
 export default function ColumnTypeLabel({ type }) {
   let stringIcon = '';
-  if (type === '' || type === 'expression') {
+  if (!type) {
+    stringIcon = '?';
+  } else if (type === '' || type === 'expression') {
```

Comment:
`type === ''` is not possible given the first if statement

Reply:
good catch I'll clarify the first statement